### PR TITLE
Contrib: add basic description of SID mem-cache

### PR DESCRIPTION
**Pull request by alexey-tikhonov**

### CODE REVIEW

File: src/contrib/mmap_cache.rst [hunk 1]
```diff
@@ -439,6 +442,38 @@ number of GIDs ``num_groups`` ``0x4`` (4). The GIDs of the groups are
...
+        uint32_t type;          /* enum sss_id_type */
+        uint32_t id;            /* gid or uid */
+        uint32_t populated_by;  /* 0 - by_id(), 1 - by_uid/gid() lookup */
+        uint32_t sid_len;       /* length of sid */
+        char sid[0];
+    };
+
+The ``name`` pointer is a shortcut to the ``sid`` data and is used to make sure
+that the object so far only found with the help of the hash value does match the
+input SID. If the user is search by ID the ``id`` value is used for this check.
```
Comment (justin-stephenson) [inline src/contrib/mmap_cache.rst:463]: The part which begins with `and is used...` and ends with `match the input SID` is a bit confusing to me. Below text sounds a bit better to me, but please validate it.

The ``name`` pointer is a shortcut to the ``sid`` data and is used to make sure that the object, found with the hash value, matches the input SID.

File: src/contrib/mmap_cache.rst [hunk 2]
```diff
@@ -439,6 +442,38 @@ number of GIDs ``num_groups`` ``0x4`` (4). The GIDs of the groups are
...
+If there is no match there is either a hash collision and the next entry in the
+chain has to be checked or the search entry is currently not in the memory cache
+and the request has to be forwarded to SSSD's nss responder.
```
Comment (justin-stephenson) [inline src/contrib/mmap_cache.rst:466]: s/request has to be/request is/

Comment (alexey-tikhonov) [reply src/contrib/mmap_cache.rst:466]: Fixed.

File: src/contrib/mmap_cache.rst (hunk 1 above)
Comment (alexey-tikhonov) [reply src/contrib/mmap_cache.rst:463]: Yes, this is better, thank you. (I just took a template from description of other types).

**Comment by alexey-tikhonov** [comment]
> Minor changes requested otherwise LGTM.

Updated.

**Review by justin-stephenson — APPROVED** [review]

**Pull request merged**